hvpy/helper.py

```diff
@@ -0,0 +1,107 @@
+import time
+from typing import Union, Optional
+from pathlib import Path
+from datetime import datetime
+
+from hvpy import downloadMovie, getMovieStatus, queueMovie
+from hvpy.api_groups.movies.queue_movie import queueMovieInputParameters
+from hvpy.utils import _add_shared_docstring, save_file
+
+__all__ = [
+    "createMovie",
+]
+
+
+@_add_shared_docstring(queueMovieInputParameters)
+def createMovie(
+    startTime: datetime,
+    endTime: datetime,
+    layers: str,
+    events: str,
+    eventsLabels: bool,
+    imageScale: float,
+    format: Optional[str] = "mp4",
+    frameRate: Optional[str] = "15",
+    maxFrames: Optional[str] = None,
+    scale: Optional[bool] = None,
+    scaleType: Optional[str] = None,
+    scaleX: Optional[float] = None,
+    scaleY: Optional[float] = None,
+    movieLength: Optional[float] = None,
+    watermark: Optional[bool] = True,
+    width: Optional[str] = None,
+    height: Optional[str] = None,
+    x0: Optional[str] = None,
+    y0: Optional[str] = None,
+    x1: Optional[str] = None,
+    y1: Optional[str] = None,
+    x2: Optional[str] = None,
+    y2: Optional[str] = None,
+    size: Optional[int] = None,
+    movieIcons: Optional[int] = None,
+    followViewport: Optional[int] = None,
+    reqObservationDate: Optional[datetime] = None,
+    overwrite: bool = False,
+    filename: Union[str, Path] = None,
+    hq: bool = False,
+) -> Path:
+    """
+    Automatically creates a movie using the endpoint `queueMovie`,
+    `getMovieStatus` and `downloadMovie`.
+
+    Parameters
+    ----------
+    {Insert}
+    overwrite
+        Whether to overwrite the file if it already exists.
+        Default is `False`.
+    filename
+        The path to save the file to.
+        Optional, will default to ``f"{starttime}_{endtime}.{format}"``.
+    """
+    input_params = locals()
+    # These are used later on but we want to avoid passing
+    # them into queueMovie.
+    input_params.pop("overwrite")
+    input_params.pop("filename")
+    input_params.pop("hq")
+
+    res = queueMovie(**input_params)
+
+    if res.get("error"):
+        raise RuntimeError(res["error"])
+
+    timeout = time.time() + 60 * 5  # 5 minutes
+    while True:
+        status = getMovieStatus(
+            id=res["id"],
+            format=format,
+            token=res["token"],
+        )
+        if status["status"] in [0, 1]:
+            time.sleep(5)
+        if status["status"] == 2:
+            break
+        if time.time() > timeout:
+            raise RuntimeError("Exceeded timeout of 5 minutes.")
+        if status["status"] == 3:
```

Comment:
Should we remove this check `(status == 3)`? 
Because this only occurs when we provide an invalid `movie_id`, in our case we directly get it from the response. So I think it is impossible to hit this line, or maybe I am wrong.

Reply:
Hmm, maybe if that's the case then. I think there isn't any harm in it being there. But we can let Daniel decide.

Reply:
But how do we maintain 100% coverage? 👀

Reply:
I don't really care if we leave it untested.

Reply:
Okay.